libcxx/test/libcxx/thread/thread.stoptoken/atomic_unique_lock.pass.cpp

```diff
@@ -12,7 +12,7 @@
 
 // UNSUPPORTED: c++03, c++11, c++14, c++17
 
-#include <__stop_token/atomic_unique_lock.h>
```

Comment:
Could we instead just add `-Wno-private-header` to all `libcxx/test/libcxx` tests? I think ideally we wouldn't export library details at all other than by including detail headers.

Reply:
We can, but I think I'd rather add it on a per test basis to avoid making it too easy to include detail headers unknowingly.